helix-lsp/src/lib.rs

```diff
@@ -619,12 +623,14 @@ impl Notification {
     }
 }
 
+type IncomingStreams = SelectAll<Pin<Box<Peekable<UnboundedReceiverStream<(usize, Call)>>>>>;
```

Comment:
Instead of `SelectAll<Peekabke` can't wu use `Peekabke<SelectAll`? That's what I had in mind and I thi k might be simpler.  ut maybe I missed something.

Reply:
Ah yeah I think you're right. I ran into some problems with the Pin when I tried that first but it looks like it's possible